# Project Euler: Additional Practice

### 3s and 5s

If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.

In [7]:
# Your Code here

list_three_five = []
for ii in range(1,1000):
    if ii%3 == 0 or ii%5 == 0:
        list_three_five.append(ii)
        
print(list_three_five)        
        
sum = 0
for ii in list_three_five:
    sum = ii +sum
    
print(sum)    

[3, 5, 6, 9, 10, 12, 15, 18, 20, 21, 24, 25, 27, 30, 33, 35, 36, 39, 40, 42, 45, 48, 50, 51, 54, 55, 57, 60, 63, 65, 66, 69, 70, 72, 75, 78, 80, 81, 84, 85, 87, 90, 93, 95, 96, 99, 100, 102, 105, 108, 110, 111, 114, 115, 117, 120, 123, 125, 126, 129, 130, 132, 135, 138, 140, 141, 144, 145, 147, 150, 153, 155, 156, 159, 160, 162, 165, 168, 170, 171, 174, 175, 177, 180, 183, 185, 186, 189, 190, 192, 195, 198, 200, 201, 204, 205, 207, 210, 213, 215, 216, 219, 220, 222, 225, 228, 230, 231, 234, 235, 237, 240, 243, 245, 246, 249, 250, 252, 255, 258, 260, 261, 264, 265, 267, 270, 273, 275, 276, 279, 280, 282, 285, 288, 290, 291, 294, 295, 297, 300, 303, 305, 306, 309, 310, 312, 315, 318, 320, 321, 324, 325, 327, 330, 333, 335, 336, 339, 340, 342, 345, 348, 350, 351, 354, 355, 357, 360, 363, 365, 366, 369, 370, 372, 375, 378, 380, 381, 384, 385, 387, 390, 393, 395, 396, 399, 400, 402, 405, 408, 410, 411, 414, 415, 417, 420, 423, 425, 426, 429, 430, 432, 435, 438, 440, 441, 444, 445, 447, 450,

### Fibonacci Sequence

Each new term in the Fibonacci sequence is generated by adding the previous two terms. By starting with 1 and 2, the first 10 terms will be:

1, 2, 3, 5, 8, 13, 21, 34, 55, 89, ...

By considering the terms in the Fibonacci sequence whose values do not exceed four million, find the sum of the even-valued terms.

In [40]:
# Your Code Here


fibo_list =[]

c1 = 1
c2 = 2
sum = c1+c2
ii = 0
fibo_list.append(c1)
fibo_list.append(c2)

print(fibo_list)

while sum < 4000000:

    sum = fibo_list[ii] +fibo_list[ii+1]
    fibo_list.append(sum)
    ii = ii+1
    print(fibo_list)
    
print(ii)  
print(sum)
print(fibo_list)
    
if fibo_list[len(fibo_list)-1] > 4000000:
    fibo_list_new = fibo_list[0:len(fibo_list)-1]


sum_total = 0
for ii in fibo_list_new:
    if ii% 2 == 0:
        sum_total =sum_total + ii
        
print(sum_total)



[1, 2]
[1, 2, 3]
[1, 2, 3, 5]
[1, 2, 3, 5, 8]
[1, 2, 3, 5, 8, 13]
[1, 2, 3, 5, 8, 13, 21]
[1, 2, 3, 5, 8, 13, 21, 34]
[1, 2, 3, 5, 8, 13, 21, 34, 55]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 1597, 2584, 4181, 6765, 10946, 17711]
[1, 2, 3, 5, 8, 13, 21, 34, 55, 89, 144, 233, 377, 610, 987, 159

### Largest Palindrome

A palindromic number reads the same both ways. The largest palindrome made from the product of two 2-digit numbers is 9009 = 91 × 99.

Find the largest palindrome made from the product of two 3-digit numbers.

In [60]:
# Your Code Here
first_number = 0
second_number = 0
temp= 0
final_product = 0

for ii in range(100,1000):
    for jj in range(100,1000):
        product = ii *jj
        product_string = str(product) 
        length = len(product_string)
        
        confirmation = []
        for item in range(length):
            if item < length/2:
                if product_string[item] == product_string[length-1-item]:
                    confirmation.append(1)
                    
        if len(confirmation) ==  length/2:
            for xx in confirmation:
                xx == 1
                if temp < product:
                
                    first_number = ii 
                    second_number = jj
                    final_product=product
                    temp = product
            

print(first_number)
print(second_number)
print(final_product)
    








913
993
906609


### Sum of Squares and Square of sums

The sum of the squares of the first ten natural numbers is,

12 + 22 + ... + 102 = 385
The square of the sum of the first ten natural numbers is,

(1 + 2 + ... + 10)2 = 552 = 3025
Hence the difference between the sum of the squares of the first ten natural numbers and the square of the sum is 3025 − 385 = 2640.

Find the difference between the sum of the squares of the first one hundred natural numbers and the square of the sum.

In [92]:
# Your Code Here
sum_square = 0
square_of_sum_temp = 0
list_number =[]

for jj in range(11,102):
    
    sum_square = 0
    square_of_sum_temp = 0
    for ii in range(jj):
        sum_square = sum_square + ii **2
        square_of_sum_temp = square_of_sum_temp + ii
    
    square_of_sum = square_of_sum_temp **2
    diff = square_of_sum - sum_square 
    list_number.append(diff)
    

print(list_number)


[2640, 3850, 5434, 7462, 10010, 13160, 17000, 21624, 27132, 33630, 41230, 50050, 60214, 71852, 85100, 100100, 117000, 135954, 157122, 180670, 206770, 235600, 267344, 302192, 340340, 381990, 427350, 476634, 530062, 587860, 650260, 717500, 789824, 867482, 950730, 1039830, 1135050, 1236664, 1344952, 1460200, 1582700, 1712750, 1850654, 1996722, 2151270, 2314620, 2487100, 2669044, 2860792, 3062690, 3275090, 3498350, 3732834, 3978912, 4236960, 4507360, 4790500, 5086774, 5396582, 5720330, 6058430, 6411300, 6779364, 7163052, 7562800, 7979050, 8412250, 8862854, 9331322, 9818120, 10323720, 10848600, 11393244, 11958142, 12543790, 13150690, 13779350, 14430284, 15104012, 15801060, 16521960, 17267250, 18037474, 18833182, 19654930, 20503280, 21378800, 22282064, 23213652, 24174150, 25164150]
